INPTP_AppForFixing/EmployeeDialog.cs

```diff
@@ -71,9 +89,30 @@ private bool CheckValidInputForm()
             int id;
             double salary;
 
-            if (Int32.TryParse(tBID.Text, out id) && Double.TryParse(tBSalary.Text, out salary) && tBID.Text.Length > 0 &&
-                tBFirstName.Text.Length > 0 && tBJob.Text.Length > 0 && tBLastName.Text.Length > 0) return true;
-            return false;
+            return Int32.TryParse(tBID.Text, out id) && 
+                Double.TryParse(tBSalary.Text, out salary) && 
+                tBID.Text.Length > 0 &&
+                tBFirstName.Text.Length > 0 && 
+                tBJob.Text.Length > 0 && 
+                tBLastName.Text.Length > 0;
+           
+        }
+
+        private bool isIdUniqui()
+        {
+            int id = Int32.Parse(tBID.Text);
+
+            foreach (var Boss in main.Bosses)
+            {
+                foreach (var Employee in Boss.GetEmployees())
+                {
+                    if (Employee.Id == id)
```

Comment:
during this check i found, that you aren't considering that you could have employee in boss in boss. This case is unhandled and could led to wrong id calculation. Also this update is similair to part of #39 pull request